### PR TITLE
test: proper integer cast in redscatbkinter.c and redscatinter.c

### LINKED ISSUE #5330: Integer overflow in redscatinter and redscatbkinter tests
redscatinter and redscatinter fail when running on more than 256 processes. 
The patch below fixes integer overflow in the test
```
diff --git a/test/mpi/coll/redscatbkinter.c b/test/mpi/coll/redscatbkinter.c
index 783006f..203d248 100644
--- a/test/mpi/coll/redscatbkinter.c
+++ b/test/mpi/coll/redscatbkinter.c
@@ -59,7 +59,7 @@ int main(int argc, char **argv)
         }

         for (i = 0; i < sendcount; i++) {
-            sendbuf[i] = (long long) (rank * sendcount + i);
+            sendbuf[i] = (long long)rank * (long long)sendcount + i;
         }
         recvbuf = (long long *) malloc(recvcount * sizeof(long long));
         if (!recvbuf) {
diff --git a/test/mpi/coll/redscatinter.c b/test/mpi/coll/redscatinter.c
index 33a04b4..b6fc936 100644
--- a/test/mpi/coll/redscatinter.c
+++ b/test/mpi/coll/redscatinter.c
@@ -69,7 +69,7 @@ int main(int argc, char **argv)
         }

         for (i = 0; i < sendcount; i++) {
-            sendbuf[i] = (long long) (rank * sendcount + i);
+            sendbuf[i] = (long long)rank * (long long)sendcount + i;
         }
         recvbuf = (long long *) malloc(recvcount * sizeof(long long));
         if (!recvbuf) {

```

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description

When cast to larger integer types, we need make sure to prevent
intermediate overflows.

Thanks github user brminich.

Fixes #5330.

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

File: test/mpi/coll/redscatbkinter.c [hunk 1]
```diff
@@ -59,7 +59,7 @@ int main(int argc, char **argv)
         }
 
         for (i = 0; i < sendcount; i++) {
-            sendbuf[i] = (long long) (rank * sendcount + i);
+            sendbuf[i] = (long long) rank *(long long) sendcount + i;
```
Comment (raffenet) [inline test/mpi/coll/redscatbkinter.c:62]: I don't think the second cast is necessary now that the parens are removed.

Comment (raffenet) [reply test/mpi/coll/redscatbkinter.c:62]: Nitpick aside this is probably fine to merge as it is.

**Review by raffenet — APPROVED** [review]

File: test/mpi/coll/redscatbkinter.c (hunk 1 above)
Comment (hzhou) [reply test/mpi/coll/redscatbkinter.c:62]: > 
> 
> I don't think the second cast is necessary now that the parens are removed.

Agreed. Will update.

**Pull request merged**